packages/amplify-category-function/src/provider-utils/awscloudformation/index.ts

```diff
@@ -217,12 +217,27 @@ export async function updateFunctionResource(context, category, service, paramet
     saveMutableState(parameters);
     saveCFNParameters(parameters);
   } else {
-    parameters = await serviceConfig.walkthroughs.updateWalkthrough(context, parameters, resourceToUpdate);
-    if (parameters.dependsOn) {
-      context.amplify.updateamplifyMetaAfterResourceUpdate(category, parameters.resourceName, 'dependsOn', parameters.dependsOn);
+    try {
+      parameters = await serviceConfig.walkthroughs.updateWalkthrough(context, parameters, resourceToUpdate);
+      if (parameters.dependsOn) {
+        context.amplify.updateamplifyMetaAfterResourceUpdate(category, parameters.resourceName, 'dependsOn', parameters.dependsOn);
+      }
+      saveMutableState(parameters);
+      saveCFNParameters(parameters);
+    } catch (e) {
+      if (e.name === 'FunctionResourceNotFound' || e.name === 'SelectedLambdaFunctionNotFound') {
+        // exit here as for these errors function parameters wont be defined
+        context.print.warning(e.message);
```

Comment:
No need to address it in this PR, but at some point, we should probably DRY our error handling throughout the codebase a bit 😄 .